src/configurationvalue.cpp

```diff
@@ -51,29 +52,30 @@ QString ConfigurationValue::key() const
     return mItem->key();
 }
 
-void ConfigurationValue::setKey(const QString &key)
+void ConfigurationValue::setKey(const QString &newKey)
 {
-    QVariant oldValue;
-
-    // don't emit valueChanged unless absolutely necessary
-    if (mItem)
-        oldValue = mItem->value(mDefaultValue);
+    QVariant oldValue = value();
+    QString oldKey = key();
 
     delete mItem;
-    mItem = new MGConfItem(key, this);
-    connect(mItem, SIGNAL(valueChanged()), this, SIGNAL(valueChanged()));
-
-    emit keyChanged();
+    if (newKey.isEmpty()) {
+        mItem = NULL;
+    } else {
+        mItem = new MGConfItem(newKey, this);
+        connect(mItem, SIGNAL(valueChanged()), this, SIGNAL(valueChanged()));
+    }
 
-    // we deleted the old item, so we must emit ourselves in this case
-    if (oldValue != mItem->value(mDefaultValue))
+    // don't emit valueChanged and keyChanged unless absolutely necessary
+    if (oldKey != key())
+        emit keyChanged();
```

Comment:
Small optimization would be to have this check at the beginning, just returning if key didn't change.